bftengine/src/preprocessor/messages/ClientBatchRequestMsg.cpp

```diff
@@ -60,6 +60,36 @@ void ClientBatchRequestMsg::validate(const ReplicasInfo& repInfo) const {
     LOG_ERROR(logger(), "Message sender is invalid" << KVLOG(senderId()));
     throw std::runtime_error(__PRETTY_FUNCTION__);
   }
+  if (!checkElements()) {
+    LOG_ERROR(logger(), "One or more ClientMsg in the list is invalid");
+    throw std::runtime_error(__PRETTY_FUNCTION__);
+  }
+}
+bool ClientBatchRequestMsg::checkElements() const {
+  const auto totalMsgSize = size();
+  const auto& numOfMessagesInBatch = msgBody()->numOfMessagesInBatch;
+  char* dataPosition = body() + sizeof(ClientBatchRequestMsgHeader) + msgBody()->cidSize;
+  auto sigManager = SigManager::instance();
+  bool isClientTransactionSigningEnabled = sigManager->isClientTransactionSigningEnabled();
+  for (auto i = 0u; i < numOfMessagesInBatch; i++) {
+    const auto& singleMsgHeader = *(ClientRequestMsgHeader*)dataPosition;
+    PrincipalId clientId = singleMsgHeader.idOfClientProxy;
+    auto expectedSigLen = (isClientTransactionSigningEnabled ? sigManager->getSigLength(clientId) : 0);
+    if (expectedSigLen != singleMsgHeader.reqSignatureLength || totalMsgSize < singleMsgHeader.requestLength ||
```

Comment:
Is it possible to put in all complex conditions brackets around? I think it makes the code more readble...and reduce amounts of errors..